src/main/java/net/minestom/server/thread/PerChunkThreadProvider.java

```diff
@@ -0,0 +1,19 @@
+package net.minestom.server.thread;
+
+import net.minestom.server.instance.Chunk;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Each {@link Chunk} gets assigned to a random thread.
+ */
+public class PerChunkThreadProvider extends ThreadProvider {
+
+    public PerChunkThreadProvider(int threadCount) {
+        super(threadCount);
+    }
+
+    @Override
+    public long findThread(@NotNull Chunk chunk) {
+        return chunk.hashCode();
```

Comment:
This isn't a random thread right?

Reply:
Indeed, otherwise the chunk thread would keep changing over time which is not ideal. I guess that I could replace it to the identity hashcode directly

Reply:
That might be better. I think not relying on the identity hash code is always good if we can.